kolibri/plugins/document_pdf_render/assets/src/views/index.vue

```diff
@@ -54,26 +80,166 @@
+            const ctx = canvas.getContext('2d');
+            const renderTask = pdfPage.render({
+              canvasContext: ctx,
+              viewport: viewport,
+            });
+            this.pdfPages[pageNum].canvas = canvas;
+            this.pdfPages[pageNum].renderTask = renderTask;
+            this.pdfPages[pageNum].rendering = true;
+            this.pdfPages[pageNum].loading = false;
+            resolve();
+            renderTask.then(
+              () => {
+                if (this.pdfPages[pageNum]) {
+                  this.pdfPages[pageNum].rendering = false;
+                  if (this.pdfPages[pageNum].canvas) {
+                    // Canvas has not been deleted in the interim
+                    this.pdfPages[pageNum].rendered = true;
+                    this.$refs[this.pageRef(pageNum)][0].appendChild(this.pdfPages[pageNum].canvas);
+                  }
+                }
+              },
+              () => {
+                if (this.pdfPages[pageNum]) {
+                  this.pdfPages[pageNum].rendering = false;
+                }
+              }
+            );
+          }
+        });
+      },
+      showPage(pageNum) {
+        if (pageNum <= this.totalPages && pageNum > 0) {
+          if (!this.pdfPages[pageNum]) {
+            this.pdfPages[pageNum] = {};
+          }
+          if (!this.pdfPages[pageNum].rendered && !this.pdfPages[pageNum].loading) {
+            this.pdfPages[pageNum].loading = true;
+            if (!this.pdfPages[pageNum].pdfPage) {
+              this.pdfPages[pageNum].displayPromise = this.getPage(pageNum).then(this.displayPage);
+            } else {
+              this.pdfPages[pageNum].displayPromise = this.displayPage(
+                this.pdfPages[pageNum].pdfPage
+              );
+            }
+          }
+        }
+      },
+      hidePage(pageNum) {
+        if (pageNum <= this.totalPages && pageNum > 0) {
+          if (!this.pdfPages[pageNum]) {
+            // No page rendered, so do nothing
+            return;
+          }
+          const pdfPage = this.pdfPages[pageNum];
+          if (pdfPage.rendered) {
+            pdfPage.rendered = false;
+            if (pdfPage.canvas) {
+              pdfPage.canvas.remove();
+            }
+          } else if (pdfPage.loading) {
+            // Currently loading, cancel the task
+            const renderTask = pdfPage.renderTask;
+            pdfPage.displayPromise.then(() => {
+              renderTask && renderTask.cancel();
+            });
+          } else if (pdfPage.rendering) {
+            // Currently rendering, cancel the task
+            pdfPage.renderTask && pdfPage.renderTask.cancel();
+          }
+          pdfPage.pdfPage && pdfPage.pdfPage.cleanup();
+          delete this.pdfPages[pageNum];
+        }
+      },
+      pageRef(index) {
+        return `pdfPage-${index}`;
+      },
+      checkPages() {
+        const top = this.$refs.pdfcontainer.scrollTop;
```

Comment:
Get current scroll position, work out which pages are visible on top and bottom, then hide and display pages appropriately.